sky/backends/backend_utils.py

```diff
@@ -2283,11 +2285,32 @@ def is_controller_up(
         if record is not None:
             controller_status, handle = record['status'], record['handle']
 
-    if controller_status is None:
-        sky_logging.print(non_existent_message)
+    error_msg = None
+    if controller_status is None or handle.head_ip is None:
```

Comment:
Q: Why do we add `handle.head_ip is None`? Actually, does it correspond to controller in STOPPED which means we should not print non_existent_message? Or should modify them: https://github.com/skypilot-org/skypilot/blob/ee811e32f8c5ae58f7cdff40392000f69e66d765/sky/utils/controller_utils.py#L89; we meant no **running** managed spot jobs / services.